presidio-analyzer/tests/test_context_support.py

```diff
@@ -90,3 +91,23 @@ def test_text_with_context_improves_score(self):
             assert(len(results_without_context) == len(results_with_context))
             for i in range(len(results_with_context)):
                 assert(results_without_context[i].score < results_with_context[i].score)
+
+    def test_text_with_context_improves_score_custom_recognizer(self):
```

Comment:
Is there a way to test the specific index problem? The test seems rather generic

Reply:
I elaborated on the special case of the test + made a unit test just for this specific part